File: 1667-FindKthBitInNthBinaryString/1667-FindKthBitInNthBinaryString.cpp
// Last updated: 3/30/2026, 2:12:11 PM
class Solution {
private:
    string reverse_toggle(string result){
        for(int index = 0 ; index < result.size() ; index++){
            if(result[index]=='1') result[index]='0';
            else result[index]='1';
        }
        reverse(result.begin(),result.end());
        return "1"+result;
    }
public:
    char findKthBit(int n, int k) {
        string res = "0";
        for(int i = 1 ; i < n ; i++){
            res+=reverse_toggle(res);
        }
        cout << res << endl;
        return res[k-1];
    }
};

// s1 = 0
// s2 = 0 + 1 + rev(toggle(0)) -> 0 + 1 + rev(1) -> 011
// s3 = 011 + 1 + rev(toggle(011)) -> 011 + 1 + rev(100) -> 0111001

File: 1667-FindKthBitInNthBinaryString/1667-FindKthBitInNthBinaryString.java
// Last updated: 3/30/2026, 2:12:16 PM
class Solution {
    public char findKthBit(int n, int k) {
        String result = "0";
        for(int ctr = 1 ; ctr < n ; ctr++){
            String reversedtoggle = reverse_toggle(result);
            result += reversedtoggle;
        }
        return result.charAt(k-1);
    }
    private String reverse_toggle(String s){
        StringBuilder sb = new StringBuilder(s);
        for(int index = 0 ; index < sb.length() ; index++){
            if(sb.charAt(index)=='0') sb.setCharAt(index,'1');
            else if(sb.charAt(index)=='1') sb.setCharAt(index,'0');
        }
        return "1"+sb.reverse().toString();
    }
}